mlflow/utils/databricks_utils.py

```diff
@@ -944,3 +945,33 @@ def get_config(self):
         # in this case, we don't need to initialize databricks token because
         # there is no backend mlflow service available.
         pass
+
+
+def get_databricks_nfs_tmp_dir():
+    entry_point = _get_dbutils().entry_point
+    if getpass.getuser() == 'ROOT':
+        return entry_point.getReplNFSTempDir()
+    else:
+        try:
+            # If it is not ROOT user, it means the code runs in Safe-spark.
+            # in this case, we should get temporary directory of current user.
+            # and `getReplNFSTempDir` will be deprecated for this case.
+            return entry_point.getUserNFSTempDir()
+        except Exception:
+            # fallback
+            return entry_point.getReplNFSTempDir()
+
+
+def get_databricks_local_tmp_dir():
+    entry_point = _get_dbutils().entry_point
+    if getpass.getuser() == 'ROOT':
+        return entry_point.getReplLocalTempDir()
+    else:
+        try:
+            # If it is not ROOT user, it means the code runs in Safe-spark.
+            # in this case, we should get temporary directory of current user.
+            # and `getReplLocalTempDir` will be deprecated for this case.
```

Comment:
`will be deprecated` means it's not deprecated?

Reply:
Will `getReplLocalTempDir` be removed in the future?

Reply:
Will getReplLocalTempDir be removed in the future?

for shared cluster (safe spark), I think so.

for assigned mode cluster, because it only has ROOT user, so `getReplLocalTempDir` will be kept otherwise all notebook kernels have to share the same temp dir.